src/Arrays/Buffers.jl

```diff
@@ -164,6 +165,42 @@ end
     end
   end
 
+
+  # CUDAdrv.jl throws on CUDA_ERROR_NOT_READY
+  function queryStream(hStream)
+      err = CUDAdrv.@runtime_ccall((:cuStreamQuery, CUDAdrv.libcuda), CUDAdrv.CUresult,
+                          (CUDAdrv.CUstream,),
+                          hStream)
+
+      if err === CUDAdrv.CUDA_ERROR_NOT_READY
+          return false
+      elseif err === CUDAdrv.CUDA_SUCCESS
+          return true
+      else
+          CUDAdrv.throw_api_error(err)
+      end
+  end
+
+  """
+      friendlysynchronize(stream)
+
+  MPI defines a notion of progress which means that MPI operations
```

Comment:
taken from https://docs.nvidia.com/deeplearning/sdk/nccl-developer-guide/docs/mpi.html#mpi-progress

Reply:
Interesting.  Thanks for the link.